### PR TITLE
[DISCOVERY-166] validate insights input file

### PR DESCRIPTION
**Pull request by bruno-fs**
- Validate insights report tarball
- Log ingress response

### CODE REVIEW

File: qpc/utils.py [hunk 1]
```diff
@@ -340,18 +340,14 @@ def setup_logging(verbosity):
     else:
         log_level = logging.DEBUG
 
+    log_fmt = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
```
Comment (nicolearagao) [inline qpc/utils.py:343]: I know that the log changes are mostly for the end user, but do you think we could benefit from adding "%(funcName)s:%(lineno)d" to formatter whenever we are on debug verbosity level?

Comment (bruno-fs) [reply qpc/utils.py:343]: makes sense

File: qpc/utils.py [hunk 2]
```diff
@@ -340,18 +340,14 @@ def setup_logging(verbosity):
...
-    # but we only adjust the log level for the 'qpc' logger.
-    log.setLevel(log_level)
-    # the StreamHandler sends warnings and above to stdout, but
-    # only for messages going to the 'qpc' logger, i.e. QPC
-    # output.
-    stderr_handler = logging.StreamHandler()
-    stderr_handler.setLevel(logging.ERROR)
-    log.addHandler(stderr_handler)
+    logging.basicConfig(filename=QPC_LOG, format=log_fmt, level=log_level)
+    stream_handler = logging.StreamHandler()
```
Comment (nicolearagao) [inline qpc/utils.py:347]: What do you think about making it a separate function?

```
def get_console_handler(log_fmt, log_level):
       setup ...
        return stream_handler
```

Comment (bruno-fs) [reply qpc/utils.py:347]: Mmm makes sense but I think this is small enough to justify not adding the "bureaucracy" of an extra function XD

File: qpc/utils.py (hunk 1 above)
Comment (bruno-fs) [reply qpc/utils.py]: any thoughts on this https://github.com/quipucords/qpc/pull/195/commits/16f36451f4cdfa1f2e34f477e11b314b5d29edcd?

File: qpc/insights/publish.py [hunk 3]
```diff
@@ -61,12 +62,63 @@ def _validate_insights_report_name(self):
...
+        except tarfile.ReadError as err:
+            log.exception(_(messages.INSIGHTS_REPORT_CONTENT_UNEXPECTED))
+            raise SystemExit(1) from err
+        return filenames
+
+    def _is_top_folder(self, top_folder):
+        expected_top_folder_parent = Path(".")
+        return top_folder.parent == expected_top_folder_parent
+
+    def _separate_top_folder(self, filenames):
```
Comment (nicolearagao) [inline qpc/insights/publish.py:95]: nice!

File: qpc/insights/tests_insights_publish.py [hunk 4]
```diff
@@ -148,7 +196,7 @@ def test_insights_publish_successful(  # pylint: disable=too-many-arguments
             "insights",
             "publish",
             "--input-file",
-            payload_file.name,
+            str(payload_file),
```
Comment (nicolearagao) [inline qpc/insights/tests_insights_publish.py:203]: why not use name attribute ?

File: qpc/utils.py (hunk 2 above)
Comment (nicolearagao) [reply qpc/utils.py:350]: Fair enough =)

File: qpc/insights/tests_insights_publish.py (hunk 4 above)
Comment (bruno-fs) [reply qpc/insights/tests_insights_publish.py:203]: it wont give the full path for the file. I didn't dig on the explanation, but the file produced by `tmp_path` fixture don't behave the same way as `NamedTemporaryFile`

Comment (bruno-fs) [reply qpc/insights/tests_insights_publish.py:203]: Now that I thought "out loud" about it makes sense to behave differently. `tmp_path` is a `pathlib.Path` object in the end :thinking:

Comment (nicolearagao) [reply qpc/insights/tests_insights_publish.py:203]: got it.

Comment (bruno-fs) [reply qpc/insights/tests_insights_publish.py:203]: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name

**Review by nicolearagao — APPROVED** [review]
LGTM 👍🏼

File: qpc/insights/publish.py [hunk 5]
```diff
@@ -61,12 +62,63 @@ def _validate_insights_report_name(self):
             log.info(_(messages.INSIGHTS_LOCAL_REPORT_NOT_TAR_GZ))
             sys.exit(1)
 
+    def _validate_insights_report_content(self):
+        """Check if report tarball contains the expected json files."""
+        filenames = self._get_filenames()
+
+        if len(filenames) < 2:
+            log.error(_(messages.INSIGHTS_REPORT_CONTENT_MIN_NUMBER))
```
Comment (gtanzillo) [inline qpc/insights/publish.py:70]: Should all of these error messages related to validation failures be prefixed with something like `Payload file is not valid, due to: <specific error message>`?

Comment (bruno-fs) [reply qpc/insights/publish.py:70]: I think it's a good idea! I'm thinking on a adding this prefix directly on directly on the messages module (https://github.com/quipucords/qpc/blob/16f36451f4cdfa1f2e34f477e11b314b5d29edcd/qpc/messages.py#L362-L370).
I think I'd have to adapt a little bit the messages to accommodate the prefix :thinking:  
What do you think?

Comment (bruno-fs) [reply qpc/insights/publish.py:70]: (the linked snippet contains only the new messages)

Comment (gtanzillo) [reply qpc/insights/publish.py:70]: Adding the prefix directly to the messages is plenty good enough for this :+1

Comment (bruno-fs) [reply qpc/insights/publish.py:70]: done!

**Review by gtanzillo — APPROVED** [review]
Looks great! Thanks @bruno-fs

**Pull request merged**